support_for_advertising_use_cases.md

```diff
@@ -529,51 +531,55 @@ This use case is common in sales period. Note that sales period can have very pr
 
 Advertisers want to optimize the look and feel of their ads to the audience and to the environment. This helps to drive more performance and improve their overall brand recognition.
 
-This can be done programmatically using historical aggregated and/or personal user data, and contextual information. 
+This can be done programmatically using historical aggregated and/or personal user data, and contextual information.
 
 ## Coupon Management
 
 As an Advertiser I want to be able to add coupons, or any form of targeted discounts to my creatives on specific period of times, potentially for specific products.
 
+# ML Applications
 
-## Recommendation
+Machine Learning Models are very important part of optimal investment in advertising. They are used by many participants in three major areas:
 
-Large e-Commerce websites may have product catalogs with millions of items. They have an interesting use-case. When running ads to promote items from their vast catalogs, which product should they show?
-Today, much of this is mainly powered by data about which items a person has previously browsed.
+## Item recommendations
 
-Typical sources of recommendation would be :
-- Historical Products (compared to historical products browsed): abandoned cart, etc.
-- Similar products (compared to historical products browsed)
-- Complementary Products (compared to historical products browsed)
-- ...
+When there are many items to choose from (products/content/messages/apps) before rendering ad, which one should be shown to which person and in which ad? This produces more useful ads for consumers. Users are less annoyed with ads because they will more likely see items they would like to buy/convert or see less products they already bought/installed or they are not interested to buy/install.
 
-# Training ML Models
+## Bidding logic
+
+Used for estimating efficiency of ad placements. When there are millions publishers to chose from and even more ad placements on those publishers with multiple available ad sizes, which one has highest chances of end conversion for advertisers and which one has highest return on investment? This drives better monetization opportunity for publishers and creates equal chances for big and small publishers to compete with performance results they do deliver for advertising campaign.
```

Comment:
I suggest a few minor grammatical changes:

"Machine Learning is used to estimate the efficacy of ad placements. Across an ad network, there might be millions of placements, across hundreds of thousands of publisher websites, each with different properties (like size and location on the page). How much return on investment will each placement drive for each advertiser? Machine learning is used to make such estimates."

Reply:
I like the above suggestion but I would add some additional words about the importance of a publisher's point of view.
 
Keeping maxim possible effective bidding logic is driving competition on the performance level between publishers and creating equal chances no matter what size of publisher you are.
For example, you can be a small, niche publisher but if you have great native experience ad placement you can compete with your ad performance next to big publishers. 
 
Or maybe we should consider to move this to the Publisher's needs?

Reply:
I would prefer to keep this document as factual as possible, statements like:

"This is the usecase. Here's an example to illustrate it. This is why it's done. This is how it is achieved today."

I would rather avoid more emotive language about what is good and bad etc. 

Perhaps we could dispassionately explain the downsides of poor-quality predictions?

From my perspective - the connection between "effective bidding logic" facilitated by ML predictions on the ad-network side, and "competition between publishers" is tenuous at best... Can you try to link the two more clearly?

Here's an example of what I think is a very factual problem one would encounter with low-quality predictions:

"The lower the accuracy of the prediction, the fewer business outcomes will be generated for the same budget. This in turn leads to lower publisher revenue."

Alternatively, we could consider it from the perspective of "model calibration". If one were to use a global average estimate of the likelihood of a conversion, and apply this same prediction to all placements, this would result in poorly calibrated campaigns. There would be some placements where the true likelihood of a conversion was much lower than the average, and bidding the average value there would result in significantly less value creation than expected."

Reply:
Noted. 

My proposal:

"The lower the accuracy of the prediction in bidding logic, the fewer business outcomes will be generated for the same budget and less precise bidding for particular ad placements may occur. This in turn leads to lower publisher revenue."

Reply:
That sounds great!

I think you could eliminate the clause "and less precise bidding for particular ad placements may occur", because "fewer business outcomes" and "lower publisher revenue" are the concrete consequences of that "less precise bidding".